src/wp-admin/includes/class-core-upgrader.php

```diff
@@ -292,38 +292,14 @@ public function upgrade( $current, $args = array() ) {
 	 * @return bool True if we should update to the offered version, otherwise false.
 	 */
 	public static function should_update_to_version( $offered_ver ) {
-		include( ABSPATH . WPINC . '/version.php' ); // $wp_version; // x.y.z
```

Comment:
The split to make this logic testable is here after this first line.

- `should_update_to_version()` just collects data - runs `include`, packages up the relevant constants into an array, and passes off to another function
- `auto_update_enabled_for_versions()` contains the core logic as a pure function (inputs map directly to outputs without depending on any other external state), and this is the function that's tested in unit tests